src/app/portlets/dot-edit-page/content/dot-edit-content.component.spec.ts

```diff
@@ -388,4 +399,22 @@ describe('DotEditContentComponent', () => {
             );
         });
     });
+
+    describe('handle switch site', () => {
+        const mockRenderedPageState = new DotRenderedPageState(mockUser, mockDotRenderedPage);
+
+        beforeEach(() => {
+            spyOn(dotPageStateService, 'get').and.returnValue(Observable.of(mockRenderedPageState));
+            component.pageState = null;
+        });
+
+        it('should rerender pagestate after switch site', () => {
+            expect(component.pageState).toBe(null);
+            fixture.detectChanges();
+
+            siteServiceMock.setFakeCurrentSite(mockSites[1]);
+            expect(dotPageStateService.get).toHaveBeenCalledWith('an/url/fake');
+            expect(component.pageState).toBe(mockRenderedPageState);
+        });
+    });
```

Comment:
We need one more test when dotPageStateService.get response with a 404 http error